typescript/nomad-deploy/src/incremental/index.ts

```diff
@@ -0,0 +1,63 @@
+import { NomadContext } from '@nomad-xyz/sdk/';
+import { canonizeId } from '@nomad-xyz/sdk/utils';
+
+/**
+ * Prepares and executes necessary calls to governing
+ * router for enrolling a spoke after core and
+ * bridge have been deployed
+ * @param sdk SDK containing new spoke domain
+ * @param spokeDomain domain of the spoke
+ * @param watchers set of watchers to be enrolled
+ */
+export async function enrollSpoke(
+  sdk: NomadContext,
+  spokeDomain: number,
+  watchers: string[],
+): Promise<void> {
+  let hubCore = await sdk.governorCore();
+  let hubBridge = sdk.mustGetBridge(hubCore.domain);
+
+  let spokeCore = await sdk.mustGetCore(spokeDomain);
+  let spokeBridge = await sdk.mustGetBridge(spokeDomain);
+  let batch = await hubCore.newGovernanceBatch();
+
+  // enroll watchers
+  await Promise.all(
+    watchers.map(async (watcher) => {
+      const call =
+        await hubCore.xAppConnectionManager.populateTransaction.watcherPermission(
+          watcher,
+          spokeDomain,
+        );
+      batch.pushLocal(call);
+    }),
+  );
+
+  // enroll replica
+  const hubReplicaOfSpoke = hubCore.getReplica(spokeDomain)?.address;
+  const enrollReplicaCall =
+    await hubCore.xAppConnectionManager.populateTransaction.ownerEnrollReplica(
+      hubReplicaOfSpoke!,
+      spokeDomain,
+    );
+  batch.pushLocal(enrollReplicaCall);
+  // set router remote
+  const setRouterCall =
+    await hubCore.governanceRouter.populateTransaction.setRouterLocal(
+      spokeDomain,
+      canonizeId(spokeCore.governanceRouter.address),
+    );
+  batch.pushLocal(setRouterCall);
+  // enroll bridge
+  const enrollBridgeCall =
+    await hubBridge.bridgeRouter.populateTransaction.enrollRemoteRouter(
+      spokeDomain,
+      canonizeId(spokeBridge.bridgeRouter.address),
+    );
+  batch.pushLocal(enrollBridgeCall);
+
+  // turn into a tx request
+  await batch.build();
+  // send to the chain
+  await batch.execute();
```

Comment:
this will only work in development. it signs and sends the tx directly to the chain. we need a solution for staging an production too (can be separate PR)

Reply:
this should be fixed in the next PR adding Gnosis support. added TODO: https://github.com/nomad-xyz/nomad-monorepo/pull/17/commits/01169d81534a0a5111b20db912c2f7cafd721852

Reply:
agree on next pr